### DISCUSSION TITLE
DistanceSensor not working

### DISCUSSION THREAD

**Discussion by jayyoder07** [gpiozero/gpiozero#990] [General]
I am using raspberrypi 3 B+ with Raspberry Pi OS Lite. I am using the example sketch listed in the docs. The sketch runs without error but there is no output.  I used my own sketch to test the and it works perfect so I know my hardware setup is working. The problem seems to be in DistanceSensor.  What am I missing here?

**Comment by bennuttall** [comment]
Need more information. What's the exact code you're running? How are you running it?

**Comment by jayyoder07** [comment]
from gpiozero import DistanceSensor
from time import sleep

sensor = DistanceSensor(17, 27)

while True:
    print('Distance to nearest object is', sensor.distance, 'm')
    sleep(1)

This is the code I'm using. It's straight out of the doc's. I'm running it in in the console. I've googled tutorials and examples for the rpi with the HC-04p. They all use the same code and they work for the authors. But none work for me. I'm kinda at my wits end trying to figure out what I'm doing wrong.

**Reply by lurch** [reply]
Can you put your code in triple-backticks so that we can see if your indentation is correct?
So just to check, the example code above is the code that **isn't** working? You also mentioned "I used my own sketch to test the and it works perfect so I know my hardware setup is working." - could you paste that code too?

**Reply by jayyoder07** [reply]
This dose not  work. This from the docs. The only change I made was the trigger and echo pins. I tried several different combinations with the same results. This gives no results. If I ad the partial=True it returns 0.0.

"""
from gpiozero import DistanceSensor
from time import sleep

sensor = DistanceSensor(17, 27, partial=True)

while True:
    print('Distance to nearest object is', sensor.distance, 'm')
    sleep(1)
"""
I don't think indentation is the problem. Python3 usually complains when that happens.



This dose work. To be clear this is not my invention. I found this somewhere on the internet. 

"""
#Libraries
import RPi.GPIO as GPIO
import time
 
#GPIO Mode (BOARD / BCM)
GPIO.setmode(GPIO.BCM)
 
def DistanceSensor(echo, trigger, queue_len):
    
    #set GPIO Pins
    GPIO_TRIGGER = trigger
    GPIO_ECHO = echo

    #set GPIO direction (IN / OUT)
    GPIO.setup(GPIO_TRIGGER, GPIO.OUT)
    GPIO.setup(GPIO_ECHO, GPIO.IN)
    
    # set Trigger to HIGH
    GPIO.output(GPIO_TRIGGER, True)
 
    # set Trigger after 0.01ms to LOW
    time.sleep(0.00001)
    GPIO.output(GPIO_TRIGGER, False)
    
    StartTime = time.time()
    StopTime = time.time()
 
    # save StartTime
    while GPIO.input(GPIO_ECHO) == 0:
        StartTime = time.time()
    
    # save time of arrival
    while GPIO.input(GPIO_ECHO) == 1:
        StopTime = time.time()
        # time difference between start and arrival
        TimeElapsed = StopTime - StartTime

    # multiply with the sonic speed (343 m/s)
    # and divide by 2, because there and back   
    distance = (TimeElapsed * 343) / 2
    
    return distance
 

if __name__ == '__main__':
    try:
        while True:
            distr = DistanceSensor(echo=6, trigger=5, queue_len=2)
            distl = DistanceSensor(echo=17, trigger=27, queue_len=2)
            print ("Measured Distance Right = ",  distr)
            print ("Measured Distance Left = ", distl)
            time.sleep(1)
 
        # Reset by pressing CTRL + C
    except KeyboardInterrupt:
        print("Measurement stopped by User")
        GPIO.cleanup()

"""
